source/server/hot_restart_impl.cc

```diff
@@ -144,11 +145,7 @@ HotRestartImpl::HotRestartImpl(Options& options)
 Stats::RawStatData* HotRestartImpl::alloc(const std::string& name) {
   // Try to find the existing slot in shared memory, otherwise allocate a new one.
   Thread::LockGuard lock(stat_lock_);
-  absl::string_view key = name;
-  if (key.size() > options_.statsOptions().maxNameLength()) {
-    key.remove_suffix(key.size() - options_.statsOptions().maxNameLength());
-  }
-  auto value_created = stats_set_->insert(key);
+  auto value_created = stats_set_->insert(truncateStatName(name));
```

Comment:
This is really nice -- maybe we could use this for the explicit `name.substr()` in `ThreadLocalStoreImpl::ScopeImpl::safeMakeStat` ?

Reply:
Yeah I don't think we need that anymore; it's done in the allocator.  Removing that (along with your TODO).